bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/handler/SunSynkInverterHandler.java

```diff
@@ -0,0 +1,479 @@
+                            5);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_6_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            6);
+                    break;
+                // Inverter control
+                case CHANNEL_INVERTER_CONTROL_TIMER:
+                    this.tempInverterChargeSettings.setPeakAndValley(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_ENERGY_PATTERN:
+                    this.tempInverterChargeSettings.setEnergyMode(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_WORK_MODE:
+                    this.tempInverterChargeSettings.setSysWorkMode(Integer.valueOf(command.toString()));
+                    break;
+            }
+
+            Optional<SunSynkAccountHandler> checkBridge = getSafeBridge();
+            if (!checkBridge.isPresent()) {
+                logger.debug("Failed to find associated SunSynk Bridge.");
+                return;
+            }
+            this.batterySettingsUpdated = true; // true = update battery settings from API at next interval
+        }
+    }
+
+    private void sendSettingsCommandToInverter() {
+        logger.debug("Ok - will handle command for CHANNEL_BATTERY_INTERVAL_1_GRID_CHARGE");
+        SunSynkInverterConfig config = getThing().getConfiguration().as(SunSynkInverterConfig.class);
+        try {
+            inverter.sendSettings(this.tempInverterChargeSettings);
+        } catch (SunSynkSendCommandException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not send command to inverter " + config.getAlias() + ". Authentication Failure !");
+            logger.debug("Could not send command to inverter {}.", config.getAlias());
+            return;
+        }
+        logger.debug("Sent command: to inverter {}.", config.getAlias());
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Running dispose()");
+        if (this.refreshTask != null) {
+            this.refreshTask.cancel(true);
+            this.refreshTask = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // config = getConfigAs(Inverter.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        logger.debug("Will boot up the inverter binding!");
+        SunSynkInverterConfig config = getThing().getConfiguration().as(SunSynkInverterConfig.class);
+        logger.debug("Inverter Config: {}", config);
+
+        if (config.getRefresh() < refreshTime) {
+            logger.warn(
+                    "Refresh time [{}] is not valid. Refresh time must be at least 60 seconds. Setting to minimum of 60 sec",
+                    config.getRefresh());
+            config.setRefresh(60);
```

Comment:
For your consideration: usually the Thing state is set to offline with  `ThingStatusDetail.CONFIGURATION_ERROR` + message and let the user fix the configuration.

Reply:
okay  done